tests/variable/variable.cpp

```diff
@@ -89,6 +89,47 @@ TEST(Variable, setUnitsAndName) {
     EXPECT_EQ(e, a);
 }
 
+TEST(Variable, setInitialValue) {
```

Comment:
Might want to rename the test ```setInitialValueString``` to make it clear that we are initialising the value using a string (as opposed to using a double).

Otherwise, rather than using variable names such as ```e```, ```v``` and ```a```, I would use more explicit names such as ```expected```, ```variable``` and ```actual```.

Note: the same is true through the whole code (it makes it easier to read/maintain).